_posts/2021-09-06-terraform-uncommon-options.markdown

```diff
@@ -0,0 +1,336 @@
+
+- Validate the JSON before pushing.
+- Increment the `serial` key by one.
+
+#### "Pull-Edit-Push"
+
+The following command will pull and write the current active state to the `terraform_state.json` file.
+
+```sh
+$ terraform state pull > terraform_state.json
+```
+
+Which has the following contents.
+
+```json
+{
+  "version": 4,
+  "terraform_version": "1.0.4",
+  "serial": 1,
+  "lineage": "b538a863-c14e-c216-9309-e5e35f91468c",
+  "outputs": {},
+  "resources": [
+    {
+      "mode": "managed",
+      "type": "local_file",
+      "name": "private_key",
+      "provider": "provider[\"registry.terraform.io/hashicorp/local\"]",
+      "instances": [
+        {
+          "schema_version": 0,
+          "attributes": {
+            "content": "{\"a\":1,\"b\":2,\"c\":3}",
+            "content_base64": null,
+            "directory_permission": "0777",
+            "file_permission": "0777",
+            "filename": "./map.json",
+            "id": "e7ec4a8f2309bdd4c4c57cb2adfb79c91a293597",
+            "sensitive_content": null,
+            "source": null
+          },
+          "sensitive_attributes": [],
+          "private": "bnVsbA=="
+        }
+      ]
+    }
+  ]
+}
+```
+
+I will remove the last element in the content parameter, and now the local state file looks like this:
+
+```json
+{
+  "version": 4,
+  "terraform_version": "1.0.4",
+  "serial": 2,
+  "lineage": "b538a863-c14e-c216-9309-e5e35f91468c",
+  "outputs": {},
+  "resources": [
+    {
+      "mode": "managed",
+      "type": "local_file",
+      "name": "private_key",
+      "provider": "provider[\"registry.terraform.io/hashicorp/local\"]",
+      "instances": [
+        {
+          "schema_version": 0,
+          "attributes": {
+            "content": "{\"a\":1,\"b\":2}",
+            "content_base64": null,
+            "directory_permission": "0777",
+            "file_permission": "0777",
+            "filename": "./map.json",
+            "id": "e7ec4a8f2309bdd4c4c57cb2adfb79c91a293597",
+            "sensitive_content": null,
+            "source": null
+          },
+          "sensitive_attributes": [],
+          "private": "bnVsbA=="
+        }
+      ]
+    }
+  ]
+}
+```
+
+After saving the file, "Pushing" the state back to its configured backend is as easy as:
+
+```sh
+$ terraform state push ./terraform_state.json
+```
+
+The "push" will fail if there's something wrong with the state file. Otherwise, it shouldn't provide any output.
+
+Further verifying that the state edit is successful can be done by running either `terraform plan` or `terraform apply`.
+
+The changes that Terraform will report attesting to the successful manual state edit.
+
+```sh
+$ terraform apply
+local_file.private_key: Refreshing state... [id=e7ec4a8f2309bdd4c4c57cb2adfb79c91a293597]
+
+Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
```

Comment:
I think "used" needs to be "uses" if you mean that Terraform generally uses selected providers for XYZ. If what you mean is "In this example, Terraform used the selected providers...", then what you wrote is correct and you can ignore this comment.

Reply:
This is output from terraform it's copied and pasted from the terminal. No need to change it.